File: backend/app/schemas/dna.py
from pydantic import BaseModel


class DnaMetric(BaseModel):
    key: str
    label: str
    value: float
    trend: float | None = None
    description: str


class TraderDnaProfile(BaseModel):
    data_status: str
    trade_count: int
    archetype: str
    risk_personality: str
    strengths: list[str]
    weaknesses: list[str]
    hidden_pnl_leaks: list[str]
    metrics: list[DnaMetric]
    behavioral_fingerprint: dict[str, float]
    message: str | None = None

File: frontend/src/lib/api/dna.ts
import { api } from '@/lib/api/client'

export interface DnaMetric {
  key: string
  label: string
  value: number
  trend?: number | null
  description: string
}

export interface TraderDnaProfile {
  data_status: 'no_trades' | 'live_analysis'
  trade_count: number
  archetype: string
  risk_personality: string
  strengths: string[]
  weaknesses: string[]
  hidden_pnl_leaks: string[]
  metrics: DnaMetric[]
  behavioral_fingerprint: Record<string, number>
  message?: string | null
}

export async function getDnaProfile(): Promise<TraderDnaProfile> {
  return api.get('/api/v1/dna/profile')
}
